tools/ci/compile_and_run.sh

```diff
@@ -11,9 +11,10 @@ cp config/example/* config/
 # Define any unit test defines that need to run
 echo "#define ${TEST_DEFINE} 1" > ${TEST_FILE}
 
-# Compile a copy of the codebase
-DreamMaker $BASENAME.dme
+# Compile a copy of the codebase, and print errors as Github Actions annotations
+DreamMaker $BASENAME.dme > compile.log
 exitVal=$?
+sed 's/^(.+?.dm):(\d+):(error|warning):(.+)$/::\3 file=\1,line=\2::\4/g' < compile.log
```

Comment:
```suggestion
sed -E 's/^(.+?.dm):(\d+):(error|warning):(.+)$/::\3 file=\1,line=\2::\4/g' < compile.log
```
backreferences need -E